grafana/influxdb_service_discovery.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2014 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
```

Comment:
I'm wondering if this is necessary. If we default to using the proxy mode,
users can easily switch to direct access via the UI. Even without the
loadbalancer discovery part, this PR makes the default user experience
fantastic.
Consider the scenario where instead of loadbalancer, K8s users end up using
node ports. Are we going to extend support for that as well?
I'm just a bit weary of maintaining more logic.

On Fri, Sep 25, 2015 at 12:37 PM, Thuc Nguyen <notifications@github.com>
wrote:

> In grafana/influxdb_service_discovery.go
> <https://github.com/kubernetes/heapster/pull/599#discussion_r40467732>:
>
> > @@ -0,0 +1,76 @@
> > +// Copyright 2014 Google Inc. All Rights Reserved.
> > +//
> > +// Licensed under the Apache License, Version 2.0 (the "License");
> > +// you may not use this file except in compliance with the License.
> > +// You may obtain a copy of the License at
> > +//
> > +//     http://www.apache.org/licenses/LICENSE-2.0
> > +//
> > +// Unless required by applicable law or agreed to in writing, software
> > +// distributed under the License is distributed on an "AS IS" BASIS,
> > +// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
> > +// See the License for the specific language governing permissions and
> > +// limitations under the License.
> > +
>
> @vishh <https://github.com/vishh> you asked a question about why this was
> needed. Grafana supports both proxy and direct methods to access InfluxDB
> backend. This is mainly for deployments where InfluxDB is exposed as an
> external service and can be accessed directly from the browser. This code
> looks up the external service URL for InfluxDB and returns it if one is
> available.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/heapster/pull/599/files#r40467732>.
>